include/envoy/upstream/cluster_manager.h

```diff
@@ -123,15 +125,15 @@ class ClusterManager {
   virtual void
   initializeSecondaryClusters(const envoy::config::bootstrap::v3::Bootstrap& bootstrap) PURE;
 
-  using ClusterInfoMap = std::unordered_map<std::string, std::reference_wrapper<const Cluster>>;
+  using ClusterInfoMap = absl::node_hash_map<std::string, std::reference_wrapper<const Cluster>>;
 
   /**
    * @return ClusterInfoMap all current clusters. These are the primary (not thread local)
    * clusters which should only be used for stats/admin.
    */
   virtual ClusterInfoMap clusters() PURE;
 
-  using ClusterSet = std::unordered_set<std::string>;
+  using ClusterSet = absl::flat_hash_set<std::string>;
```

Comment:
are we sure this is safe?

Reply:
Could you define safe? Yes we don't see regressions. The contract of unordered_set is that the order is indeterminate. We've already shown up that the hash values constructed by msvc Cary from those of gcc or clang. We know from the test changes that many coders take an easy way out with the treatment of unordered data sets, and the abesil team has gone to great lengths by adding entroy to prove up such foolish assumptions. Yes, the patch is ready to accept and merge. Where exceptions can be thrown by elemental constructors, we have to deal with those, but we think we've found them all. That's a legit bug that abesil should address and already has an open ticket. As far as predictability of behavior of envoy based on unordered associations in end user configs... It seems time to bite the bullet and merge this, and then introduce still-breaking changes by introducing contracted behavioral ordering based on sorted collections as needed. 1.15 just dropped, so the time to begin evaluating this for 1.16 on a wide scale was yesterday.

Reply:
I'm not asking about the whole of this PR, was just asking why this is the only place that you chose absl::flat_hash_set over absl::node_hash_set. Did you checked all usage of this type that doesn't rely on pointer stability?

Reply:
We moved to `flat_hash_set` to try to reduce memory usage that the stats integration tests showed was increased because of this change (see this thread: https://github.com/envoyproxy/envoy/pull/11879#discussion_r450429690)

I believe I checked if this set was used w/o a need for pointer stability but ill look again, we haven't seen any test coverage show anything i believe yet

Reply:
Thanks for clarification, SGTM.